components/rpc/invoker/mosn/channel/xchannel.go

```diff
@@ -93,8 +97,104 @@ type xChannel struct {
 	pool  *connPool
 }
 
-// Do is handle RPCRequest to RPCResponse
-func (m *xChannel) Do(req *rpc.RPCRequest) (*rpc.RPCResponse, error) {
+// InvokeWithTargetAddress send request to specific provider address
+func (m *xChannel) InvokeWithTargetAddress(req *rpc.RPCRequest) (*rpc.RPCResponse, error) {
```

Comment:
why not deliver this header to MOSN， and let MOSN do the connection logic？

Reply:
> why not deliver this header to MOSN， and let MOSN do the connection logic？
Haven't seen you for a long time, for this question:

1. MOSN doesn't have this logic now。
2. If i add an stream filter, i need to add this stream filter to the configuration file, which may affect the existing mosn logic. That is, the impact area will become larger。
3. Maybe one day layotto will no longer be based on mosn。

By the way , I will pool the direct connection, it is a good choice to let rpc support multiple channels。

Reply:
I had the same concerns as @MoonShining 
but
>3. Maybe one day layotto will no longer be based on mosn。

I think it's  a good point. 
So how about we implement it outside the `mosninvoker`, for example, in a new `directInvoker`?
With this design, if one day users want to switch to envoy or some other rpc framework, this logic still works